internal/controllers/cryostat_controller.go

```diff
@@ -397,6 +406,59 @@ func (r *CryostatReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return c.Complete(r)
 }
 
+func (r *CryostatReconciler) reconcileReports(ctx context.Context, reqLogger logr.Logger, instance *operatorv1beta1.Cryostat,
+	routeTLS *openshiftv1.TLSConfig, imageTags *resources.ImageTags, serviceSpecs *resources.ServiceSpecs) (reconcile.Result, error) {
+	reqLogger.Info("Spec", "Reports", instance.Spec.ReportOptions)
+
+	if instance.Spec.ReportOptions == nil {
+		instance.Spec.ReportOptions = &operatorv1beta1.ReportConfiguration{Replicas: 0}
+	}
+	desired := instance.Spec.ReportOptions.Replicas
+
+	if desired == 0 {
+		svc := resources.NewReportService(instance)
+		if err := r.Client.Delete(ctx, svc); err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		deployment := resources.NewDeploymentForReports(instance, imageTags)
+		if err := r.Client.Delete(ctx, deployment); err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if desired > 0 {
+		svc := resources.NewReportService(instance)
+		if err := controllerutil.SetControllerReference(instance, svc, r.Scheme); err != nil {
+			return reconcile.Result{}, err
+		}
+		if err := r.createObjectIfNotExists(ctx, types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}, &corev1.Service{}, svc); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		deployment := resources.NewDeploymentForReports(instance, imageTags)
+		if err := controllerutil.SetControllerReference(instance, deployment, r.Scheme); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		podTemplate := deployment.Spec.Template.DeepCopy()
+		op, err := controllerutil.CreateOrUpdate(ctx, r.Client, deployment, func() error {
+			deployment.Spec.Template.Spec = podTemplate.Spec
+			deployment.Spec.Replicas = &desired
+			return nil
+		})
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		serviceSpecs.ReportsURL = &url.URL{
+			Scheme: "http",
```

Comment:
We should create issue(s) to set up secure traffic between Cryostat and the report generator using cert-manager.